js/src/components/account-card/index.js

```diff
@@ -75,33 +81,33 @@ const alignStyleName = {
  *
  * @param {Object} props React props.
  * @param {string} [props.className] Additional CSS class name to be appended.
- * @param {APPEARANCE | {icon, title, defaultDescription}} props.appearance Kind of account to indicate the card appearance, or a tuple with icon, title and optional defaultDescription to be used.
  * @param {boolean} [props.disabled=false] Whether display the Card in disabled style.
- * @param {JSX.Element} [props.description] Content below the card title. It will fall back to `appearance.defaultDescription` if not specified and the default is applicable.
+ * @param {APPEARANCE} [props.appearance=APPEARANCE.EMPTY]
+ *   Kind of account to indicate the default card appearance, which could include icon, title, and description.
+ *   If didn't specify this prop, all properties of appearance will be `undefined` (nothing shown),
+ *   and it's usually used for full customization.
+ * @param {JSX.Element} [props.icon] Card icon. It will fall back to the icon of respective `appearance` config if not specified.
+ * @param {JSX.Element} [props.title] Card title. It will fall back to the title of respective `appearance` config if not specified.
+ * @param {JSX.Element} [props.description] Description content below the card title. It will fall back to the description of respective `appearance` config if not specified.
  * @param {JSX.Element} [props.helper] Helper content below the card description.
- * @param {boolean} [props.hideIcon=false] Whether hide the leading icon.
- * @param {'center'|'top'} [props.alignIcon='center'] Specify the vertical alignment of leading icon.
  * @param {JSX.Element} [props.indicator] Indicator of actions or status on the right side of the card.
+ * @param {'center'|'top'} [props.alignIcon='center'] Specify the vertical alignment of leading icon.
  * @param {'center'|'top'} [props.alignIndicator='center'] Specify the vertical alignment of `indicator`.
  * @param {Array<JSX.Element>} [props.children] Children to be rendered if needs more content within the card.
  */
 export default function AccountCard( {
 	className,
-	appearance,
 	disabled = false,
-	description,
+	appearance = 'empty',
+	icon = appearanceDict[ appearance ].icon,
+	title = appearanceDict[ appearance ].title,
+	description = appearanceDict[ appearance ].description,
```

Comment:
❔ 💅 This relates back to my comment in https://github.com/woocommerce/google-listings-and-ads/pull/1102#discussion_r753686922 about "on the consumer perspective, it starts to feel a bit funny because the consumer can provide description via `description` props, or `appearance` props like `appearance={{ description: '' }}`."

I just thought about this idea: what about if we make those `APPEARANCE.GOOGLE` an individual object in their own file, like this:

```js
// file: appearances/google.js
// let consumer get access to the appearance object, 
// instead of appearance string.
export default {
	icon: googleLogo,
	title: __( 'Google Ads', 'google-listings-and-ads' ),
	description: __(
		'Required to create paid campaigns with your product listings',
		'google-listings-and-ads'
	),
};
```

Then, on the consumer side, we can do this:

```js
// import the object from above.
import appearanceGoogle from './appearances/google.js'

// use it via `appearance` prop.
// with custom description.
// don't need icon, title, description props.
const card1 = <AccountCard
	appearance={ { ...appearanceGoogle, description: 'Hello!' } }
/>

// Or use it via AccountCard props: icon, title, description.
// with custom description.
// don't need appearance props.
// Personally, I'm leaning towards this.
const card2 = <AccountCard
	{ ... appearanceGoogle }
	description='Hello!'
/>
```

With the above approach, we only need `appearance` props, or the "icon, title, description" props. Also, we can remove `appearanceDict` too.

Reply:
> This relates back to my comment in https://github.com/woocommerce/google-listings-and-ads/pull/1102#discussion_r753686922 about "on the consumer perspective, it starts to feel a bit funny because the consumer can provide description via `description` props, or `appearance` props like `appearance={{ description: '' }}`."

I thought your former concern was there were two methods to set card `description` and it brought some confusion. By this PR, it changed to have only one way to specify the `description` from the consumer side, and the `appearance` was changed to specify the default and highly reused card appearance only. The concern itself should have already been addressed?

> I just thought about this idea: what about if we make those `APPEARANCE.GOOGLE` an individual object in their own file
[...]
With the above approach, we only need `appearance` props, or the "icon, title, description" props. Also, we can remove `appearanceDict` too.

I don't see the `appearanceDict` causing any problems, and I'm not sure about the benefits of making them standalone files for the present. In addition, using `appearance` with props object spreading in every `<AccountCard>` makes them more tedious to read and develop. In my opinion, under the premise of high reuse base props, internalizing them is relatively simple and easy to use. And it already supports the `card2` method as well.

Reply:
> the `appearance` was changed to specify the default and highly reused card appearance only.

Sorry, my bad, I just realized the `appearance` prop has been changed from accepting an object to now accepting a string. 🤦 😅